trie/patriciaMerkleTrie.go

```diff
@@ -691,6 +694,39 @@ func (tr *patriciaMerkleTrie) GetTrieStats(address string, rootHash []byte) (*st
 	return ts.GetTrieStats(), nil
 }
 
+// CollectLeavesForMigration will collect trie leaves that need to be migrated. The leaves are collected in the trieMigrator.
+// The traversing of the trie is done in a DFS manner, and it will stop when the gas runs out (this will be signaled by the trieMigrator).
+func (tr *patriciaMerkleTrie) CollectLeavesForMigration(
+	oldVersion core.TrieNodeVersion,
+	newVersion core.TrieNodeVersion,
+	trieMigrator vmcommon.DataTrieMigrator,
+) error {
+	tr.mutOperation.Lock()
+	defer tr.mutOperation.Unlock()
+
+	if check.IfNil(tr.root) {
+		return nil
+	}
+	if check.IfNil(trieMigrator) {
+		return errors.ErrNilTrieMigrator
+	}
+
+	if newVersion > core.MaxValidTrieNodeVersion || oldVersion > core.MaxValidTrieNodeVersion {
```

Comment:
this will raise backward-compatible issues when we will have another implementation. Suggestion to completely replace the core.MaxValidTrieNodeVersion with a specialized component that is epoch-aware.

Reply:
Done.